platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,215 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.configuration;
+
+import static org.osgi.service.cm.ConfigurationEvent.CM_DELETED;
+import static org.osgi.service.cm.ConfigurationEvent.CM_UPDATED;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import org.codice.ddf.admin.configurator.Configurator;
+import org.codice.ddf.admin.configurator.ConfiguratorException;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.configurator.Operation;
+import org.codice.ddf.admin.configurator.OperationReport;
+import org.codice.ddf.admin.configurator.Result;
+import org.codice.ddf.internal.admin.configurator.actions.ConfigActions;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.cm.ConfigurationEvent;
+import org.osgi.service.cm.SynchronousConfigurationListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Addresses shortcomings of the Felix {@code ConfigInstaller} with our own implementation that properly
+ * syncs the etc directory using the {@code felix.fileinstall.filename} property. Handling a config delete
+ * is fairly straight forward. The following outcomes are possible when an update occurs:
+ * <p>
+ * <table>
+ * <tr>
+ * <td>felixFileOriginal</td><td>felixFileNew</td><td>comments</td>
+ * </tr>
+ * <tr>
+ * <td>null</td><td>null</td><td>Felix file prop wasn't tracked and is still not tracked</td>
+ * </tr>
+ * <tr>
+ * <td>null</td><td>X</td><td>New config with a new Felix file prop to track</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>null</td><td>Felix file prop no longer being tracked, delete config file and remove from map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>X</td><td>Felix file prop updated, delete config file and update map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>Y</td><td>Felix file prop did not change, do nothing</td>
+ * </tr>
+ * </table>
+ *
+ * @see #handleUpdate(ConfigurationEvent)
+ * @see #handleDelete(ConfigurationEvent)
+ */
+public class ConfigurationInstaller implements SynchronousConfigurationListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationInstaller.class);
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    private final ConfiguratorFactory configuratorFactory;
+
+    private final ConfigActions configActions;
+
+    private Map<String, File> pidFileMap;
```

Comment:
✏️ Following on the comment from @paouelle and expanding...
Since init is only called once at startup and since we now have a thread-safe map anyway...my suggestion is

1. Make the map final
2. In your `init` method, instead of using `Collectors.toConcurrentMap`, change to just calling `toMap` (still with a merge function to deal with potential key clashes) and wrap in a call to `pidFileMap.putAll()`.